Src/FluentAssertions/Primitives/ObjectAssertions.cs

```diff
@@ -16,6 +16,97 @@ public ObjectAssertions(object value)
+            .FailWith("Expected {context} to be {0}{reason}, but found {1}.", expected,
+                Subject);
+
+        return new AndConstraint<ObjectAssertions>(this);
+    }
+
+    /// <summary>
+    /// Asserts that a value does not equal <paramref name="unexpected"/> using the provided <paramref name="comparer"/>.
+    /// </summary>
+    /// <param name="unexpected">The unexpected value</param>
+    /// <param name="comparer">
+    /// An equality comparer to compare values.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="comparer"/> is <see langword="null"/>.</exception>
+    public AndConstraint<ObjectAssertions> NotBe<TExpectation>(TExpectation unexpected, IEqualityComparer<TExpectation> comparer, string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(comparer);
+
+        Execute.Assertion
+            .ForCondition(Subject is not TExpectation subject || !comparer.Equals(subject, unexpected))
+            .BecauseOf(because, becauseArgs)
+            .WithDefaultIdentifier(Identifier)
+            .FailWith("Did not expect {context} to be equal to {0}{reason}.", unexpected);
+
+        return new AndConstraint<ObjectAssertions>(this);
+    }
+
+    /// <summary>
+    /// Asserts that a value is one of the specified <paramref name="validValues"/> using the provided <paramref name="comparer"/>.
+    /// </summary>
+    /// <param name="validValues">
+    /// The values that are valid.
+    /// </param>
+    /// <param name="comparer">
+    /// An equality comparer to compare values.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])"/> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="validValues"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentNullException"><paramref name="comparer"/> is <see langword="null"/>.</exception>
+    public AndConstraint<ObjectAssertions> BeOneOf<TExpectation>(IEnumerable<TExpectation> validValues,
+        IEqualityComparer<TExpectation> comparer,
+        string because = "",
+        params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(validValues);
```

Comment:
🤔 What if the collection is empty? I saw that the other overload also doesn't check for this. But wouldn't it be strange if you call this method without any expected values?

Reply:
I added tests to verify that the assertion fails if an empty enumerable is provided.